### PR TITLE
Report: Debtor Group Annual Report

### PR DESCRIPTION
**Pull request by jniles**
This PR adds the debtor group annual report.  It's best described in screenshots. 

**With Data**
debtorgroupannualreport

**Without Data**
debtorgroupannualreporterror

### CODE REVIEW

File: client/src/partials/reports_proposed/debtor_group/annual.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          <button class="btn btn-default"
+            ng-disabled="AnnualCtrl.state === 'loading'"
+            ng-click="AnnualCtrl.generateDocument(AnnualReportForm.$invalid)"
+            ng-if="!AnnualCtrl.generatedDocumentPath">
+              {{ AnnualCtrl.state === 'loading' ? 'Loading' : 'Generate Document' }}
+          </button>
+
+          <a class="btn btn-success" ng-href="{{ AnnualCtrl.generatedDocumentPath }}" target="_blank"
+            ng-if="AnnualCtrl.generatedDocumentPath" ng-click="AnnualCtrl.clearPath()">
+            Document Build Success - Open Document
```
Comment (DedrickEnc) [inline client/src/partials/reports_proposed/debtor_group/annual.html]: This text is not translated

File: client/src/partials/reports_proposed/debtor_group/annual.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* changed here as well.
+*/
+function DebtorGroupAnnualReportController($http, $sce) {
+  var vm = this;
+
+  vm.state = 'fresh';
+
+  // TODO -- is this defined in a central location somewhere?  Maybe a service?
+  // Configuration objects optionally passed to /report/build - drives configuration UI
+  vm.configuration = {
```
Comment (DedrickEnc) [inline client/src/partials/reports_proposed/debtor_group/annual.js]: There is service called ```reportConfigService``` it provides data like configuration and this service must be used by all pdf report until now.

File: client/src/partials/reports_proposed/debtor_group/balance.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      })
+      .error(function (code) {
+        session.building = false;
+
+        // TODO Handle error
+      });
+    }
+
+    // Utility method - GET PDF blob displaying embedded object
+    function downloadDocument(url) {
```
Comment (DedrickEnc) [inline client/src/partials/reports_proposed/debtor_group/balance.js]: two functions called ```downloadDocument``` and both of them are not used, they must be removed

File: client/src/partials/reports_proposed/debtor_group/annual.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    $http.post(path, configurationObject)
+    .then(function (response) {
+      vm.generatedDocumentPath = response.data;
+    })
+    .catch(handler)
+    .finally(function () { vm.state = 'done'; });
+  }
+
+  // Utility method - GET PDF blob displaying embedded object
+  function downloadDocument(url) {
```
Comment (DedrickEnc) [inline client/src/partials/reports_proposed/debtor_group/annual.js]: the functions called ```downloadDocument``` is not used, it must be removed

File: client/src/partials/reports_proposed/debtor_group/balance.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+.controller('configureBalance', [
+  '$scope',
+  '$http',
+  '$sce',
+  'validate',
+  function ($scope, $http, $sce, validate) {
+
+    var dependencies = {};
+
+    // Configuration objects optionally passed to /report/build - drives configuration UI
```
Comment (DedrickEnc) [inline client/src/partials/reports_proposed/debtor_group/balance.js]: same comment than the pevious page

File: server/controllers/reports_proposed/data/debtor_group_annual_report.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+      });
+
+      return totals;
+    }, aggregates);
+
+    // format everything in USD
+    Object.keys(context.accounts).forEach(function (key) {
+      var account = context.accounts[key];
+
+      Object.keys(account).forEach(function (k) {
```
Comment (DedrickEnc) [inline server/controllers/reports_proposed/data/debtor_group_annual_report.js]: Good idea

File: server/controllers/reports_proposed/templates/debtor_group_annual_report.dot [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  <link rel="stylesheet" href="{{=it.path}}/templates/style/skeleton.css">
+  <link rel="stylesheet" href="{{=it.path}}/templates/style/reports.css">
+  <meta charset="utf-8" />
+</head>
+
+<body>
+
+  <hgroup class="text-uppercase">
+    <h6>IMCK</h6>
+    <h6>Service de Comptabilite</h6>
```
Comment (DedrickEnc) [inline server/controllers/reports_proposed/templates/debtor_group_annual_report.dot]: These text must be translated

File: server/lib/db.js [hunk 8]
```diff
@@ -53,11 +53,13 @@ function exec(sql, params) {
...
     if (err) { return defer.reject(err); }
 
-    connection.query(sql, params, function (err, results) {
+    var qs = connection.query(sql, params, function (err, results) {
       connection.release();
       if (err) { return defer.reject(err); }
       defer.resolve(results);
     });
+
+    console.log(qs.sql);
```
Comment (DedrickEnc) [inline server/lib/db.js]: is this ```console.log``` important?

**Comment by DedrickEnc** [comment]
I like the page, it is funcy. Also this way of handling empty dataset

File: client/src/partials/reports_proposed/debtor_group/balance.js (hunk 5 above)
Comment (jniles) [inline client/src/partials/reports_proposed/debtor_group/balance.js]: Curiously, I have no memory of writing any of the debtor_group balance code.  I'll delete this file and all the other balance duplicates.  No idea how that happened, possibly a broken merge.

File: server/lib/db.js (hunk 8 above)
Comment (jniles) [inline server/lib/db.js]: Nope, that was debug information.  I'll remove it.

File: client/src/partials/reports_proposed/debtor_group/annual.js (hunk 2 above)
Comment (jniles) [inline client/src/partials/reports_proposed/debtor_group/annual.js]: Sure.  I think when I started writing this report it hadn't been merged yet.  I'll check it out.

**Comment by jniles** [comment]
@dedrickenc, this is ready for another round of testing.

**Pull request merged**